pkg/sql/parser/sql.y

```diff
@@ -3477,14 +3477,23 @@ show_csettings_stmt:
   }
 | SHOW CLUSTER SETTING ALL
   {
-    $$.val = &tree.ShowAllClusterSettings{}
+    $$.val = &tree.ShowClusterSettingList{All: true}
   }
 | SHOW CLUSTER error // SHOW HELP: SHOW CLUSTER SETTING
 | SHOW ALL CLUSTER SETTINGS
   {
-    $$.val = &tree.ShowAllClusterSettings{}
+    $$.val = &tree.ShowClusterSettingList{All: true}
   }
 | SHOW ALL CLUSTER error // SHOW HELP: SHOW CLUSTER SETTING
+| SHOW CLUSTER SETTINGS
```

Comment:
would it make sense, for completeness, to have a `SHOW INTERNAL CLUSTER SETTINGS` counterpart to `SHOW PUBLIC CLUSTER SETTINGS`?

Reply:
I am not too fond of it. Specialized syntactic variants are intended for frequently-used cases. This is (hopefully) a rare case, which has a separate syntax already: `select * from [show all cluster settings] where public is false`.